csharp/Ice/interceptor/AuthenticatorI.cs

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+using Demo;
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+
+class AuthenticatorI : AuthenticatorDisp_
+{
+    internal AuthenticatorI()
+    {
+        _tokenStore = new Dictionary<string, long>();
+        _rand = new RNGCryptoServiceProvider();
+    }
+
+    public override string getToken(Ice.Current current)
+    {
+        //
+        // Generate a random 32 character long token.
+        //
+        string chars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        char[] tokenValue = new char[32];
+        byte[] bytes = new byte[tokenValue.Length];
+        _rand.GetBytes(bytes);
+
+        for(var i = 0; i < bytes.Length; i++)
+        {
+            tokenValue[i] = chars[bytes[i] % chars.Length];
+        }
+
+        string token = new string(tokenValue);
+        //
+        // By default tokens are valid for 1 hour after being issued.
+        //
+        long expireTime = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() + (1000 * 60 * 60);
```

Comment:
Like for C++, it would be good to check if there's a better way to compare times and add offsets in C#.